dask/array/creation.py

```diff
@@ -143,3 +144,36 @@ def arange(*args, **kwargs):
         elem_count += bs
 
     return Array(dsk, name, chunks, dtype=dtype)
+
+
+def diag(v):
+    """Construct a diagonal array, with ``v`` on the diagonal.
+
+    TODO: implement diagonal extraction, and the ``k``th diagonal keyword arg.
```

Comment:
Suggest removing the TODO from linspace as well

On Tue, Jun 9, 2015 at 9:22 PM, Jim Crist <notifications@github.com> wrote:

> In dask/array/creation.py
> <https://github.com/ContinuumIO/dask/pull/288#discussion_r32059084>:
>
> > @@ -143,3 +144,36 @@ def arange(*args, **kwargs):
> >          elem_count += bs
> >
> >      return Array(dsk, name, chunks, dtype=dtype)
> > +
> > +
> > +def diag(v):
> > +    """Construct a diagonal array, with ``v`` on the diagonal.
> > +
> > +    TODO: implement diagonal extraction, and the ``k``th diagonal keyword arg.
>
> I just put it in there because there was one in linspace already. I
> agree, and can remove it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/dask/pull/288/files#r32059084>.
>